File: HomeWork/HW1/src/main/java/model/person/Person.java
package model.person;

import model.person.visitor.PersonVisitorIfc;

/**
 * @author Haim Adrian
 * @since 08-Nov-20
 */
public abstract class Person {
    private final String id;
    private final String name;

    public Person(String id, String name) {
        this.id = id;
        this.name = name;
    }

    public String getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public void introduce() {
        System.out.println(toString());
    }

    public abstract <R> R accept(PersonVisitorIfc<R> visitor);

    @Override
    public String toString() {
        return "id='" + id + '\'' + ", name='" + name + "'";
    }
}

File: ClassWork/CW9Kotlin/src/main/kotlin/Person.kt
/**
 *
 * @author Haim Adrian
 * @since 30-May-21
 */
open class Person(var name: String = "", age: Int = 20, var salary: Int = 5500) {
    var age: Int = 0
        set(value) {
            field = if (value >= 18)
                value
            else
                18
        }

    init {
        this.age = age
    }

    override fun toString(): String {
        return "name='$name', age=$age, salary=$salary"
    }
}
